test/upload/signature_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+ describe 'upload-signature' do
+   it 'returns 400 with nothing to sign' do
+     resp = Excon.get(CONFIG.UPLOAD_HOST)
```

Comment:
Might be good to explicitly test this with an empty `to_sign=` param, or possibly both